Userland/Games/Flood/Board.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Board.h"
+#include <AK/Random.h>
+#include <time.h>
+
+Board::Board(size_t rows, size_t columns)
+{
+    resize(rows, columns);
+}
+
+Board::Board(Board const& board)
+    : m_rows(board.m_rows)
+    , m_columns(board.m_columns)
+    , m_current_color(board.m_current_color)
+    , m_previous_color(board.m_current_color)
+    , m_colors(board.m_colors)
+    , m_cells(board.m_cells)
+{
+}
+
+void Board::clear()
+{
+    for (size_t row = 0; row < m_rows; ++row) {
+        for (size_t column = 0; column < m_columns; ++column) {
+            set_cell(row, column, Color::Transparent);
+        }
+    }
+}
+
+bool Board::is_flooded() const
+{
+    Color first_cell_color = cell(0, 0).release_value();
+    for (size_t row = 0; row < rows(); ++row) {
+        for (size_t column = 0; column < columns(); ++column) {
+            if (cell(row, column).release_value() == first_cell_color)
+                continue;
+            return false;
+        }
+    }
+    return true;
+}
+
+void Board::randomize()
+{
+    for (size_t row = 0; row < m_rows; ++row) {
+        for (size_t column = 0; column < m_columns; ++column) {
+            Color const& color = m_colors[get_random<u32>() % m_colors.size()];
```

Comment:
Be careful, this does not give you uniform random if `m_colors.size()` is not a power of 2